docs/migrations/25-10.md

```diff
@@ -78,6 +78,36 @@ In the type system, queue channels are represented as `Channel`, while value cha
 Nextflow supports Groovy-style type annotations using the `<type> <name>` syntax, but this approach is deprecated in {ref}`strict syntax <strict-syntax-page>`. While Groovy-style annotations remain valid for functions and local variables, the language server and `nextflow lint` automatically convert them to Nextflow-style annotations during code formatting.
 :::
 
+See {ref}`migrating-static-types` to get started.
+
+<h3>Typed processes</h3>
+
+Typed processes are a new way to define process inputs and outputs with type annotations:
+
+```nextflow
+process fastqc {
+    input:
+    id: String
+    fastq_1: Path
+    fastq_2: Path?
+
+    output:
+    logs = tuple(id, file('fastqc_logs'))
+
+    script:
+    """
+    mkdir fastqc_logs
+    fastqc -o fastqc_logs -f fastq -q ${fastq_1} ${fastq_2}
+    """
+}
+```
+
+The new `stage:` and `topic:` sections can be used to describe additional process-specific behaviors such as custom file names and topic emissions.
```

Comment:
It might be worth adding an example to show these and adding links to the reference documentation for these.